tests/autodiff/loop-init.slang

```diff
@@ -0,0 +1,54 @@
+//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj -output-using-type
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<float> outputBuffer;
+
+struct A : IDifferentiable
```

Comment:
Should we add some file-check test lines to ensure the intermediate context doesn't have the undesired fields?

Reply:
Added 👍